helpme/views.py

```diff
@@ -26,17 +26,35 @@ def form_invalid(self, form):
             return response
 
     def form_valid(self, form):
-        form.instance.category = self.category
         form.instance.user = self.request.user
         form.instance.site = Site.objects.get_current()
+
+        user_agent = get_user_agent(self.request)
```

Comment:
This might save you some cycles since it's already added to the request object by the middleware.

`user_agent = self.request.user_agent`